attalos/dataset/dataset.py

```diff
@@ -67,27 +87,21 @@ def get_next_batch(self, batch_size):
 
         # For each item to extract from the batch
         for i, item_index in enumerate(items_in_batch):
-            # Transform image index to image_id
-            item_id = self.image_ids[item_index]
-            # Make sure our id's are strings
-            if not isinstance(item_id, str):
-                item_id = str(item_id)
+            img_feat, text_feat = self.get_index(item_index)
 
             # Add image features to output
-            img_feats[i, :] = self.image_feats[item_index, :]
-
-            # If we are supposed to transform tags using TextTransformer then transform tags
-            if self.text_feat_type == 'tags' and self.tag_transformer is not None:
-                text_feat = []
-                for feat in self.text_feats[item_id]:
-                    feat = str(feat)
-                    text_feat.append(self.tag_transformer[feat])
-            else:
-                text_feat = self.text_feats[item_id]
+            img_feats[i, :] = img_feat
             text_feats.append(text_feat)
         return img_feats, text_feats
 
+    def get_num_imgs(self):
```

Comment:
There is an argument to be made (for example: https://stackoverflow.com/questions/17330160/how-does-the-property-decorator-work/17330273#17330273) that `@property` is preferable to getters and setters, but I have no strong opinion on it.